### ISSUE TITLE
API endpoint not found error with voxbox backend and voice model

### ISSUE THREAD

**Issue by UGhost-X** [gpustack/gpustack#3522]
### GPUStack version

v2.0.0

### ▶️ Steps to reproduce

<img width="2997" height="1925" alt="Image" src="https://github.com/user-attachments/assets/2919fb54-5691-4f42-8420-50afab57e279" />

<img width="893" height="981" alt="Image" src="https://github.com/user-attachments/assets/063a7fd2-f0a6-4540-8223-8aefe4f37456" />

### ❌ Actual result

2025-11-25T08:02:08+08:00 - root - INFO - New registry table added: preprocessor_classes
/usr/local/lib/python3.12/dist-packages/jieba/__init__.py:44: SyntaxWarning: invalid escape sequence '\.'
  re_han_default = re.compile("([\u4E00-\u9FD5a-zA-Z0-9+#&\._%\-]+)", re.U)
/usr/local/lib/python3.12/dist-packages/jieba/__init__.py:46: SyntaxWarning: invalid escape sequence '\s'
  re_skip_default = re.compile("(\r\n|\s)", re.U)
/usr/local/lib/python3.12/dist-packages/jieba/finalseg/__init__.py:78: SyntaxWarning: invalid escape sequence '\.'
  re_skip = re.compile("([a-zA-Z0-9]+(?:\.\d+)?%?)")
/usr/local/lib/python3.12/dist-packages/funasr/models/fsmn_kws/encoder.py:65: SyntaxWarning: invalid escape sequence '\['
  splits = line.strip().strip('\[\]').strip().split()
/usr/local/lib/python3.12/dist-packages/funasr/models/fsmn_kws/encoder.py:124: SyntaxWarning: invalid escape sequence '\['
  splits = line.strip().strip('\[\]').strip().split()
/usr/local/lib/python3.12/dist-packages/funasr/models/fsmn_kws/encoder.py:135: SyntaxWarning: invalid escape sequence '\['
  splits = bias_line.strip().strip('\[\]').strip().split()
/usr/local/lib/python3.12/dist-packages/funasr/models/fsmn_kws/encoder.py:256: SyntaxWarning: invalid escape sequence '\['
  params_split = params_line.strip().strip('\[\]').strip().split()
/usr/local/lib/python3.12/dist-packages/funasr/models/fsmn_kws/encoder.py:277: SyntaxWarning: invalid escape sequence '\['
  splits = line.strip().strip('\[\]').strip().split()
/usr/local/lib/python3.12/dist-packages/funasr/models/fsmn_kws/encoder.py:300: SyntaxWarning: invalid escape sequence '\['
  splits = line.strip().strip('\[\]').strip().split()
<unknown>:65: SyntaxWarning: invalid escape sequence '\['
<unknown>:124: SyntaxWarning: invalid escape sequence '\['
<unknown>:135: SyntaxWarning: invalid escape sequence '\['
<unknown>:256: SyntaxWarning: invalid escape sequence '\['
<unknown>:277: SyntaxWarning: invalid escape sequence '\['
<unknown>:300: SyntaxWarning: invalid escape sequence '\['
2025-11-25T08:02:10+08:00 - root - INFO - New registry table added: adaptor_classes
2025-11-25T08:02:11+08:00 - root - INFO - New registry table added: lid_predictor_classes
/usr/local/lib/python3.12/dist-packages/jamo/jamo.py:79: SyntaxWarning: invalid escape sequence '\w'
  hcj_name = re.sub("(?<=HANGUL )(\w+)",
/usr/local/lib/python3.12/dist-packages/jamo/jamo.py:210: SyntaxWarning: invalid escape sequence '\w'
  jamo_name = re.sub("(?<=HANGUL )(\w+)",
/usr/local/lib/python3.12/dist-packages/funasr/utils/kws_utils.py:10: SyntaxWarning: invalid escape sequence '\s'
  symbol_str = '[’!"#$%&\'()*+,-./:;<>=?@，。?★、…【】《》？“”‘’！[\\]^_`{|}~\s]+'
2025-11-25T08:02:11+08:00 - root - INFO - download models from model hub: ms
2025-11-25T08:02:13+08:00 - root - INFO - Loading pretrained params from /home/nfs_share/model/SenseVoiceSmall/model.pt
2025-11-25T08:02:13+08:00 - root - INFO - ckpt: /home/nfs_share/model/SenseVoiceSmall/model.pt
2025-11-25T08:03:16+08:00 - root - INFO - scope_map: ['module.', 'None']
2025-11-25T08:03:16+08:00 - root - INFO - excludes: None
2025-11-25T08:03:17+08:00 - root - INFO - Loading ckpt: /home/nfs_share/model/SenseVoiceSmall/model.pt, status: <All keys matched successfully>
2025-11-25T08:03:17+08:00 - vox_box.server.server - INFO - Starting Vox Box server.
2025-11-25T08:03:17+08:00 - vox_box.server.server - INFO - Serving on 192.168.37.220:40011.
2025-11-25T08:04:19+08:00 - python_multipart.multipart - WARNING - Skipping data after last boundary
2025-11-25T08:04:26+08:00 - python_multipart.multipart - WARNING - Skipping data after last boundary
2025-11-25T09:03:51+08:00 - python_multipart.multipart - WARNING - Skipping data after last boundary

**Comment by aiwantaozi** [comment]
Hi, I tried deploying ModelScope’s iic/SenseVoiceSmall on an NVIDIA GPU, but I wasn’t able to reproduce the issue. Could you try deploying iic/SenseVoiceSmall in your environment to see if it works on your side?

If you can share more details about your setup, we’d be happy to try reproducing the problem again.

**Comment by UGhost-X** [comment]
> Hi, I tried deploying ModelScope’s iic/SenseVoiceSmall on an NVIDIA GPU, but I wasn’t able to reproduce the issue. Could you try deploying iic/SenseVoiceSmall in your environment to see if it works on your side?
> 
> If you can share more details about your setup, we’d be happy to try reproducing the problem again.

这个修好了，原来现在模型类型不会自动探测了，还是用了默认的llm选项导致了错误，现在还有一个也有问题

1、模型用的是 modelscope download --model iic/CosyVoice-300M


<img width="923" height="1781" alt="Image" src="https://github.com/user-attachments/assets/4863c98a-72ff-415b-b140-246c1410617b" />

2、后台没有报错：
2025-11-25 09:00:57.812269+08:00 - 51606 - gpustack_runtime.deployer.docker - WARNING - Mirrored deployment enabled, but no Container name set, using hostname(kubesphere-master) instead
2025-11-25 09:00:57.894158+08:00 - 51606 - gpustack_runtime.deployer.docker - INFO - Mirrored deployment enabled, using self Container ca0bb68d597babee1b1cca22fc868f12881526371ce375930dfed250bdc6c4cf for options mirroring
2025-11-25 09:00:58.627320+08:00 - gpustack.worker.backends.vox_box - INFO - Created Vox-Box container workload cosyvoice-300m-C7I0K
2025-11-25 09:00:58.627522+08:00 - gpustack.worker.serve_manager - INFO - Finished provisioning model instance cosyvoice-300m-C7I0K
2025-11-25T09:01:08+08:00 - vox_box.cmd.start - INFO - Starting with arguments: [('debug', None), ('host', '192.168.37.220'), ('port', 40009), ('model', '/home/nfs_share/model/CosyVoice-300M'), ('device', 'cuda:0'), ('huggingface_repo_id', None), ('model_scope_model_id', None), ('data_dir', '/var/lib/gpustack'), ('func', <function run at 0x7f233f2065c0>)]
2025-11-25T09:01:08+08:00 - vox_box.server.model - INFO - Estimating model
2025-11-25T09:01:09+08:00 - vox_box.server.model - INFO - Finished estimating model
2025-11-25T09:01:09+08:00 - vox_box.server.model - INFO - Loading model
/usr/local/lib/python3.12/dist-packages/diffusers/models/lora.py:393: FutureWarning: `LoRACompatibleLinear` is deprecated and will be removed in version 1.0.0. Use of `LoRACompatibleLinear` is deprecated. Please switch to PEFT backend by installing PEFT: `pip install peft`.
  deprecate("LoRACompatibleLinear", "1.0.0", deprecation_message)
2025-11-25T09:01:21+08:00 - root - INFO - input frame rate=50
/usr/local/lib/python3.12/dist-packages/torch/nn/utils/weight_norm.py:143: FutureWarning: `torch.nn.utils.weight_norm` is deprecated in favor of `torch.nn.utils.parametrizations.weight_norm`.
  WeightNorm.apply(module, name, dim)
/usr/local/lib/python3.12/dist-packages/vox_box/backends/tts/../../third_party/CosyVoice/cosyvoice/dataset/processor.py:24: UserWarning: torchaudio._backend.set_audio_backend has been deprecated. With dispatcher enabled, this function is no-op. You can remove the function call.
  torchaudio.set_audio_backend('soundfile')
/usr/local/lib/python3.12/dist-packages/onnxruntime/capi/onnxruntime_inference_collection.py:123: UserWarning: Specified provider 'CUDAExecutionProvider' is not in available provider names.Available providers: 'AzureExecutionProvider, CPUExecutionProvider'
  warnings.warn(
2025-11-25T09:01:45+08:00 WETEXT INFO building fst for zh_normalizer ...
2025-11-25T09:01:45+08:00 - wetext-zh_normalizer - INFO - building fst for zh_normalizer ...
2025-11-25T09:02:09+08:00 WETEXT INFO done
2025-11-25T09:02:09+08:00 - wetext-zh_normalizer - INFO - done
2025-11-25T09:02:09+08:00 WETEXT INFO fst path: /usr/local/lib/python3.12/dist-packages/tn/zh_tn_tagger.fst
2025-11-25T09:02:09+08:00 - wetext-zh_normalizer - INFO - fst path: /usr/local/lib/python3.12/dist-packages/tn/zh_tn_tagger.fst
2025-11-25T09:02:09+08:00 WETEXT INFO           /usr/local/lib/python3.12/dist-packages/tn/zh_tn_verbalizer.fst
2025-11-25T09:02:09+08:00 - wetext-zh_normalizer - INFO -           /usr/local/lib/python3.12/dist-packages/tn/zh_tn_verbalizer.fst
2025-11-25T09:02:09+08:00 WETEXT INFO found existing fst: /usr/local/lib/python3.12/dist-packages/tn/en_tn_tagger.fst
2025-11-25T09:02:09+08:00 - wetext-en_normalizer - INFO - found existing fst: /usr/local/lib/python3.12/dist-packages/tn/en_tn_tagger.fst
2025-11-25T09:02:09+08:00 WETEXT INFO                     /usr/local/lib/python3.12/dist-packages/tn/en_tn_verbalizer.fst
2025-11-25T09:02:09+08:00 - wetext-en_normalizer - INFO -                     /usr/local/lib/python3.12/dist-packages/tn/en_tn_verbalizer.fst
2025-11-25T09:02:09+08:00 WETEXT INFO skip building fst for en_normalizer ...
2025-11-25T09:02:09+08:00 - wetext-en_normalizer - INFO - skip building fst for en_normalizer ...
2025-11-25T09:02:43+08:00 - vox_box.server.server - INFO - Starting Vox Box server.
2025-11-25T09:02:43+08:00 - vox_box.server.server - INFO - Serving on 192.168.37.220:40009.
2025-11-25T09:06:02+08:00 - vox_box.utils.log - INFO - Starting speech method in CosyVoice
2025-11-25T09:06:03+08:00 - vox_box.utils.log - INFO - Starting speech method in CosyVoice
2025-11-25T09:06:06+08:00 - vox_box.utils.log - INFO - Starting speech method in CosyVoice


3、页面没有返回转换结果，不知道那里中断了

<img width="2891" height="1771" alt="Image" src="https://github.com/user-attachments/assets/1b7ee9fc-3578-47fc-b42c-72b9ae24de33" />

**Comment by aiwantaozi** [comment]
You can try the following steps to check whether the UI can properly generate the speaker list.

Currently, the built-in voices file in the ModelScope repository [have been removed](https://modelscope.cn/models/iic/CosyVoice-300M/files/21630e45770fab75691ea2cc8b9f5ce9c02775b1), which may cause generation errors.
If this is the issue, you can try the following workaround:

1. Download the [voices file](https://github.com/gpustack/vox-box/blob/main/vox_box/backends/tts/cosyvoice_spk2info.pt).
2. Rename it to `spk2info.pt`.
3. Place it inside your model directory /home/nfs_share/model/CosyVoice-300M.
4. Redeploy the model using the same directory.

After redeploying, check whether the voice list appears and whether audio generation works as expected.

If this is not the issue, please open the browser’s **F12 Developer Tools** and check the API response to see what error is returned.

**Comment by UGhost-X** [comment]
> You can try the following steps to check whether the UI can properly generate the speaker list.
> 
> Currently, the built-in voices file in the ModelScope repository [have been removed](https://modelscope.cn/models/iic/CosyVoice-300M/files/21630e45770fab75691ea2cc8b9f5ce9c02775b1), which may cause generation errors. If this is the issue, you can try the following workaround:
> 
> 1. Download the [voices file](https://github.com/gpustack/vox-box/blob/main/vox_box/backends/tts/cosyvoice_spk2info.pt).
> 2. Rename it to `spk2info.pt`.
> 3. Place it inside your model directory /home/nfs_share/model/CosyVoice-300M.
> 4. Redeploy the model using the same directory.
> 
> After redeploying, check whether the voice list appears and whether audio generation works as expected.
> 
> If this is not the issue, please open the browser’s **F12 Developer Tools** and check the API response to see what error is returned.

好像是我用的模型是他的最新版弄出来的问题，换了版本现在是可以了，只是现在dify的这个gpustack插件，现在用这个注册这个模型会提示错误，不知道这个插件是不是你们这里维护耶，好久没更新了

<img width="1215" height="82" alt="Image" src="https://github.com/user-attachments/assets/29469452-f5b7-4f61-8bea-bc9867716077" />

**Comment by gitlawr** [comment]
Track the Dify plugin issue in https://github.com/gpustack/gpustack/issues/3595
